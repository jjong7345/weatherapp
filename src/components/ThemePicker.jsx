import React, { Component } from "react";
import { connect } from "react-redux";
import styled, { keyframes } from "styled-components";

import { setTheme } from "../actions";

const ThemeBlocksContainer = styled.div`
	width: 180px;
	height: 150px;
	border-radius: 10px;
	overflow: hidden;
	background: #000;
	margin: 10px auto;
	flex-flow: row wrap;
	flex: 0 0 auto;
	display: flex;
	padding: 0;
`;

const ThemeBlock = styled.div`
	width: 60px;
	height: 50px;
	cursor: pointer;
`;

class ThemePicker extends Component {
	constructor(props) {
		super(props);
		this.state = {
			themes: [
				"#C34D40",
				"#4242BA",
				"#2E3C56",
				"#E59C2F",
				"#412F3F",
				"#EA724C",
				"#5C2533",
				"#2D442F",
				"#8DD397"
			]
		};
	}
	render() {
		let content = this.state.themes.map((item, index) => (
			<ThemeBlock
				onClick={() => this.props.switchTheme(item)}
				style={{ background: item }}
				key={index}
			/>
		));
		return <ThemeBlocksContainer>{content}</ThemeBlocksContainer>;
	}
}

const mapStateToProps = state => ({
	theme: state.theme
});

const mapDispatchToProps = dispatch => ({
	switchTheme(theme) {
		dispatch(setTheme(theme));
	}
});

export default connect(mapStateToProps, mapDispatchToProps)(ThemePicker);
