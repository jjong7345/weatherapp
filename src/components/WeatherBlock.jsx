import React, { Component } from "react";
import styled, { keyframes } from "styled-components";
import { H_Box, H_Box_Align_Center } from "../style.js";

const ForcastBlock = styled.div`
	width: 100%;
	max-width: 200px;
	text-align: center;
	padding: 30px;
`;

const NowBlock = styled.div`
	width: 100%;
	max-width: 200px;
	text-align: center;
	padding: 10px;
`;

const WeatherIcon = styled.div`
	width: 100%;
	text-align: center;
	font-size: 50px;
`;

const WeatherIconNow = styled.div`
	width: 100%;
	text-align: center;
	font-size: 160px;
	padding-bottom: 10px;
`;

const High = styled.div`
	font-weight: 800;
	font-size: 30px;
	padding: 20px 5px;
`;

const Low = styled.div`
	font-weight: 800;
	font-size: 25px;
	opacity: 0.5;
	padding: 15px 5px;
`;

const Day = styled.div`
	font-weight: 100;
	font-size: 30px;
	opacity: 0.5;
`;

const Today = styled.div`
	font-weight: 100;
	font-size: 50px;
	opacity: 0.5;
`;

const Condition = styled.div`
	font-weight: 100;
	font-size: 16px;
	opacity: 0.5;
	height: 40px;
`;

const ConditionNow = styled.div`
	font-weight: 100;
	font-size: 30px;
	opacity: 0.5;
`;

export const WeatherBlock = class WeatherBlock extends Component {
	render() {
		return (
			<ForcastBlock>
				<WeatherIcon className={"wi wi-yahoo-" + this.props.code} />
				<H_Box_Align_Center>
					<High>{this.props.highTemp + "°"}</High>
					<Low>{this.props.lowTemp + "°"}</Low>
				</H_Box_Align_Center>
				<Day>{this.props.day}</Day>
				<Condition>{this.props.condition}</Condition>
			</ForcastBlock>
		);
	}
};

export const WeatherBlockNow = class WeatherBlockNow extends Component {
	render() {
		return (
			<NowBlock>
				<WeatherIconNow className={"wi wi-yahoo-" + this.props.code} />
				<High>{this.props.temp + "°"}</High>
				<Today>Now</Today>
				<ConditionNow>{this.props.condition}</ConditionNow>
			</NowBlock>
		);
	}
};
