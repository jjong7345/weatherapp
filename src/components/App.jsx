import React, { Component } from "react";
import { connect } from "react-redux";
import styled, { css } from "styled-components";
import { slide as Menu } from "react-burger-menu";

import {
	getForeCastData,
	getCurrentData
} from "../actions";

import {
	H_Box,
	H_Box_Center,
	H_Box_Align_Center,
	h_box_base
} from "../style.js";

import { WeatherBlock, WeatherBlockNow } from "./WeatherBlock";
import ThemePicker from "./ThemePicker";

const ForeCastContainer = styled.div`
	${h_box_base} padding: 30px 0;
	max-width: 800px;
	width: 100%;
	flex-flow: row wrap;
`;

const MenuTitle = styled.div`
	font-size: 20px;
	color: #000;
	font-weight: 800;
	margin: 5px 0;
`;

const CityInput = styled.input`
	font-size: 20px;
	color: #000;
	font-weight: 100;
	font-style: italic;
	margin: 15px 0 25px;
	width: 100%;
    background: transparent;
    border: none;
    border-bottom: 1px solid red;
`;

const CityName = styled.div`
	font-size: 18px;
	color: #FFF;
	font-weight: 800;
	width: 100%;
	text-align: center;
	opacity: 0.7;
`;

class App extends Component {
	constructor(props) {
		super(props);
	}

	componentDidMount() {
		this.props.getFCastData();
	}

	render() {
		let content;
		if (this.props.forecastData) {
			let condition = this.props.forecastData.query.results.channel.item
				.condition;
			let forecast = this.props.forecastData.query.results.channel.item
				.forecast;
			let forecastContent = forecast
				.slice(0, 4)
				.map((item, index) => (
					<WeatherBlock
						code={item.code}
						highTemp={item.high}
						lowTemp={item.low}
						day={item.day}
						condition={item.text}
						key={index}
					/>
				));
			content = (
				<H_Box_Align_Center>
					<WeatherBlockNow
						code={condition.code}
						temp={condition.temp}
						condition={condition.text}
					/>
					<ForeCastContainer>{forecastContent}</ForeCastContainer>
				</H_Box_Align_Center>
			);
		}
		return (
			<div className="main" style={{ background: this.props.theme }}>
				<Menu>
					<MenuTitle>Select a City</MenuTitle>
					<CityInput value="New York, NY" onChange={() => { return false }} />
					<MenuTitle>Theme</MenuTitle>
					<ThemePicker />
				</Menu>
				<div className="content">{content}</div>
				<CityName>New York, NY</CityName>
			</div>
		);
	}
}

const mapStateToProps = state => ({
	forecastData: state.forecastData,
	theme: state.theme
});

const mapDispatchToProps = dispatch => ({
	getFCastData() {
		dispatch(getForeCastData());
	}
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
