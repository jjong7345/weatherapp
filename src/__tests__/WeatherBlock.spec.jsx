import React from "react";
import Enzyme from "enzyme";
import { shallow } from "enzyme";
import { WeatherBlock } from "../components/WeatherBlock";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

test("WeatherBlock renders correctly", () => {
	const component = shallow(<WeatherBlock code="20" highTemp="70" lowTemp="50" day="Thur" condition="Cloudy"/>);
	expect(component).toMatchSnapshot();
});
