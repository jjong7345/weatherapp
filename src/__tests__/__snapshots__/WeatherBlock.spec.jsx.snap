// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherBlock renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherBlock
    code="20"
    condition="Cloudy"
    day="Thur"
    highTemp="70"
    lowTemp="50"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div
          className="wi wi-yahoo-20"
        />,
        <styled.div>
          <styled.div>
            70°
          </styled.div>
          <styled.div>
            50°
          </styled.div>
        </styled.div>,
        <styled.div>
          Thur
        </styled.div>,
        <styled.div>
          Cloudy
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "wi wi-yahoo-20",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              70°
            </styled.div>,
            <styled.div>
              50°
            </styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "70°",
            },
            "ref": null,
            "rendered": "70°",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "50°",
            },
            "ref": null,
            "rendered": "50°",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Thur",
        },
        "ref": null,
        "rendered": "Thur",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Cloudy",
        },
        "ref": null,
        "rendered": "Cloudy",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div
            className="wi wi-yahoo-20"
          />,
          <styled.div>
            <styled.div>
              70°
            </styled.div>
            <styled.div>
              50°
            </styled.div>
          </styled.div>,
          <styled.div>
            Thur
          </styled.div>,
          <styled.div>
            Cloudy
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "wi wi-yahoo-20",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                70°
              </styled.div>,
              <styled.div>
                50°
              </styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "70°",
              },
              "ref": null,
              "rendered": "70°",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "50°",
              },
              "ref": null,
              "rendered": "50°",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Thur",
          },
          "ref": null,
          "rendered": "Thur",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Cloudy",
          },
          "ref": null,
          "rendered": "Cloudy",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
